### PR TITLE
Draft PR

### PR DESCRIPTION
**Pull request by christopher-hakkaart**

### CODE REVIEW

File: docs/nf_templates/4_anatomy.md [hunk 1]
```diff
@@ -0,0 +1,37 @@
+# Anatomy of the nf-core template
+
+The nf-core pipeline template comes packed with a lot of files and folders which can be overwhelming.
+
+Fortunately, a complete understanding isn't required to get started.
+
+<figure class="excalidraw">
+--8<-- "docs/nf_templates/img/4_template.excalidraw.svg"
```
Comment (ewels) [inline docs/nf_templates/4_anatomy.md:8]: I find this pretty difficult to look at. Suggest to either remove, or have as a "photo of a screen" type thing, where it's (a) smaller and (b) clear that the reader isn't meant to interpret anything it says.

Comment (ewels) [reply docs/nf_templates/4_anatomy.md:8]: Even an image of the terminal or something could work, eg:

codetoimg-snippet (1)
codetoimg-snippet

Comment (christopher-hakkaart) [reply docs/nf_templates/4_anatomy.md:8]: Yeah, I see your point. I'll swap it out with a screenshot.

File: docs/nf_templates/7_testing.md [hunk 2]
```diff
@@ -0,0 +1,73 @@
...
+The nf-core template pipeline also comes with profiles for the management of software dependencies (e.g., `docker`, `singularity`, & `conda`). For pipelines with modules that have been shipped with containers/images/recipes, these profiles can be used to change the way dependencies are handled when you execute your pipeline.
+
+Using the `test` profile with one of the profiles for managing your software dependencies the nf-core template pipeline will execute two processes, `FASTQC` and `MULTIQC`.
+
+!!! question "Exercise"
+
+    Execute your pipeline template using the `test` and `docker` profiles using the following command.
+
+    ```bash
+    nextflow run nf-core-mypipeline -profile test,docker --outdir results
```
[the remark this answers is not in the document]
Comment (mribeirodantas) [reply docs/nf_templates/7_testing.md:20]: The MultiQC process is failing in my Gitpod instance due to the container image 🤔 

Is it working over there?

```
ERROR ~ Error executing process > 'MYORG_MYPIPELINE:MYPIPELINE:MULTIQC'

Caused by:
  Process `MYORG_MYPIPELINE:MYPIPELINE:MULTIQC` terminated with an error exit status (125)

Command executed:

  multiqc \
      --force \
       \
      --config multiqc_config.yml \
       \
       \
      .
  
  cat <<-END_VERSIONS > versions.yml
  "MYORG_MYPIPELINE:MYPIPELINE:MULTIQC":
      multiqc: $( multiqc --version | sed -e "s/multiqc, version //g" )
  END_VERSIONS

Command exit status:
  125

Command output:
  (empty)

Command error:
  Unable to find image 'quay.io/biocontainers/multiqc:1.21--pyhdfd78af_0' locally
  1.21--pyhdfd78af_0: Pulling from biocontainers/multiqc
  93e509ffbdaf: Pulling fs layer
  820c67f471a9: Pulling fs layer
  bd9ddc54bea9: Pulling fs layer
  1591479aec4e: Pulling fs layer
  1591479aec4e: Waiting
  bd9ddc54bea9: Download complete
  820c67f471a9: Verifying Checksum
  820c67f471a9: Download complete
  93e509ffbdaf: Download complete
  93e509ffbdaf: Pull complete
  1591479aec4e: Verifying Checksum
  1591479aec4e: Download complete
  docker: failed to register layer: lsetxattr user.overlay.impure /etc: operation not supported.
  See 'docker run --help'.

Work dir:
  /workspace/gitpod/docs/nf_templates/myorg-mypipeline/work/56/ee1196bb3a01e4b7986ce077c032fd

Tip: view the complete command output by changing to the process work dir and entering the command `cat .command.out`

 -- Check '.nextflow.log' file for details
```

File: docs/nf_templates/7_testing.md [hunk 3]
```diff
@@ -0,0 +1,73 @@
...
+
+The `nextflow.config` would need to be updated to include it as an optional profile.
+
+```console title="nextflow.config"
+test      { includeConfig 'conf/test.config'      }
+test2     { includeConfig 'conf/test2.config'     }
+test_full { includeConfig 'conf/test_full.config' }
+```
+
+Finally, you can customise CI pipeline run tests as required.
```
Comment (mribeirodantas) [inline docs/nf_templates/7_testing.md:61]: Changing to make it consistent with other uses of customize with a z.
```suggestion
Finally, you can customize CI pipeline run tests as required.
```

File: docs/nf_templates/8_components.md [hunk 4]
```diff
@@ -0,0 +1,167 @@
+# nf-core components
+
+The nf-core ecosystem includes a repository of components. These components are written and maintained by the nf-core community. Components are written for diverse users and may require additional configuration to suit different use cases. There are two types of components, modules and subworkflows:
+
+-   **Modules** are wrappers around single process that contain tests, environments, and meta data.
```
Comment (mribeirodantas) [inline docs/nf_templates/8_components.md:5]:
```suggestion
-   **Modules** are wrappers around a single process that contains tests, environments, and metadata.
```

File: docs/nf_templates/8_components.md [hunk 5]
```diff
@@ -0,0 +1,167 @@
...
+nf-core modules install
+```
+
+You can follow the prompts to find and install the module you are interested in:
+
+```console
+? Tool name: fastp
+```
+
+Once selected, the tooling will install the module in `modules/nf-core/` folder and give you a suggested line of code for your main workflow file (e.g., `workflows/mypipeline.nf`).
```
Comment (mribeirodantas) [inline docs/nf_templates/8_components.md:41]:
```suggestion
Once selected, the tooling will install the module in the `modules/nf-core/` folder and give you a suggested line of code for your main workflow file (e.g., `workflows/mypipeline.nf`).
```

File: docs/nf_templates/8_components.md [hunk 6]
```diff
@@ -0,0 +1,167 @@
...
+INFO     Use the following statement to include this module:
+
+include { FASTP } from '../modules/nf-core/fastp/main'
+```
+
+!!! question "Exercise"
+
+    Install the `fastp` module to your pipeline.
+
+!!! note "What is fastp?
```
Comment (mribeirodantas) [inline docs/nf_templates/8_components.md:54]:
```suggestion
!!! note "What is fastp?"
```

File: docs/nf_templates/8_components.md [hunk 7]
```diff
@@ -0,0 +1,167 @@
...
+
+    Fastp is a...
+
+### Adding a module to your pipeline
+
+Although the modules has been installed in your repository, it is not yet added to your pipeline. The suggested `include` statement needs to be added to your `workflows/mypipeline.nf` file.
```
Comment (mribeirodantas) [inline docs/nf_templates/8_components.md:60]:
```suggestion
Although the module has been installed in your repository, it is not yet added to your pipeline. The suggested `include` statement needs to be added to your `workflows/mypipeline.nf` file.
```

File: docs/nf_templates/8_components.md [hunk 8]
```diff
@@ -0,0 +1,167 @@
...
+
+```console
+include { FASTP } from '../modules/nf-core/fastp/main'
+```
+
+!!! question "Exercise"
+
+    Add the `include { FASTP } from '../modules/nf-core/fastp/main'` statement to your `workflows/mypipeline.nf` file.
+
+To implement the module in your pipeline you will need to check what inputs are required. You can view the inputs channels for the module by opening the `/modules/nf-core/fastp/main` file.
```
Comment (mribeirodantas) [inline docs/nf_templates/8_components.md:70]:
```suggestion
To implement the module in your pipeline you will need to check what inputs are required. You can view the input channels for the module by opening the `/modules/nf-core/fastp/main` file.
```

File: docs/nf_templates/8_components.md [hunk 9]
```diff
@@ -0,0 +1,167 @@
...
+
+```groovy title="/modules/nf-core/fastp/main.nf"
+input:
+tuple val(meta), path(reads)
+path  adapter_fasta
+val   save_trimmed_fail
+val   save_merged
+```
+
+Each module has a meta file which describes the required inputs. The meta file is rendered on the [nf-core website](https://nf-co.re/modules/fastp), or can be viewed using the `nf-core modules info command`.
```
Comment (mribeirodantas) [inline docs/nf_templates/8_components.md:80]:
```suggestion
Each module has a `meta.yml` file which describes the required inputs. This meta file is rendered on the [nf-core website](https://nf-co.re/modules/fastp), or can be viewed using the `nf-core modules info command`.
```

File: docs/nf_templates/8_components.md [hunk 10]
```diff
@@ -0,0 +1,167 @@
...
+
+Code for `workflows/mypipeline.nf`
```
Comment (mribeirodantas) [inline docs/nf_templates/8_components.md:82]: Reminder to fill this out with the code.

File: docs/nf_templates/8_components.md [hunk 11]
```diff
@@ -0,0 +1,167 @@
...
+    false,
+    false
+)
+```
+
+!!! note "What is `[]`?"
+
+    Every channel needs to be filled for a process to run. In the example above `[]` was used to fill the channel with an empty list.
+
+Importantly, nf-core components are tracked in `modules.json`. Tracking components improves reproducibility and reporting by giving some certainty about the version of the tool you were using. The `modules.json` will be updated with a hash for the module.
```
Comment (mribeirodantas) [inline docs/nf_templates/8_components.md:104]:
```suggestion
Importantly, nf-core components are tracked in the `modules.json` file. Tracking components improves reproducibility and reporting by giving some certainty about the version of the tool you were using. The `modules.json` file will be updated with a hash for the installed module.
```

File: docs/nf_templates/8_components.md [hunk 12]
```diff
@@ -0,0 +1,167 @@
...
+    Private repositories of modules can be accessed using `--git-remote` in `nf-core modules install` command.
+
+### Linting modules
+
+The `nf-core lint` command will help manage nf-core components and test that they match the remote they came from. Although modules are written to be flexible, you may want to modify a component to fit your purpose.
+
+For example, if you modify an nf-core module, it will no longer match the remote and a linting test of this module would fail.
+
+```
+EXAMPLE
```
Comment (mribeirodantas) [inline docs/nf_templates/8_components.md:117]: Reminder to fill this out with the example.

File: docs/nf_templates/8_components.md [hunk 13]
```diff
@@ -0,0 +1,167 @@
...
+│ Module name  │ File path                     │ Test message                               │
+│╶─────────────┼───────────────────────────────┼────────────────────────────────────────────│
+│ fastp        │ modules/nf-core/fastp/main.nf │ Local copy of module does not match remote │
+│              ╵                               ╵                                            │
+╰───────────────────────────────────────────────────────────────────────────────────────────╯
+```
+
+Changing a module does not mean you can't continue to use that module.
+
+The nf-core modules patch command allows you keep using the nf-core component without needing to make it into a local module and curate it yourself. Instead, it creates a path file that will keep track of the changes you made. If you subsequently update the module using the nf-core tooling, the diff file will be retained. If any subsequent changes to the module conflict with your diff file, you will prompted to resolve the conflicts.
```
Comment (mribeirodantas) [inline docs/nf_templates/8_components.md:142]:
```suggestion
The nf-core modules patch command allows you keep using the nf-core component without needing to make it into a local module and curate it yourself. Instead, it creates a path file that will keep track of the changes you made. If you subsequently update the module using the nf-core tooling, the diff file will be retained. If any subsequent changes to the module conflict with your diff file, you will be prompted to resolve the conflicts.
```

File: docs/nf_templates/9_sync.md [hunk 14]
```diff
@@ -0,0 +1,32 @@
+# TEMPLATE syncs
+
+The template evolves as the ecosystem evolves.
+
+New templates are release semi-regularly and the nf-core tooling helps incorporate these changes through syncs with the `TEMPLATE` branch.
```
Comment (mribeirodantas) [inline docs/nf_templates/9_sync.md:5]:
```suggestion
New templates are released semi-regularly and the nf-core tooling helps incorporate these changes through syncs with the `TEMPLATE` branch.
```

File: docs/nf_templates/10_bump.md [hunk 15]
```diff
@@ -0,0 +1,33 @@
...
+
+```bash
+git add .
+git commit -m "Version 1.0 release"
+git push
+```
+
+## Update minimum Nextflow version
+
+The template also includes a minimum Nextflow version that is required for the pipeline to to. You can also change the required version of Nextflow using the `nf-core bump-version` command.
```
Comment (mribeirodantas) [inline docs/nf_templates/10_bump.md:21]:
```suggestion
The template also includes a minimum Nextflow version that is required for the pipeline to run. You can also change the required version of Nextflow using the `nf-core bump-version` command.
```

**Review by mribeirodantas — APPROVED** [review]
Very nice material, Chris! 👏 

I left a few suggestions. Let me know if you need any help, or have questions.

File: docs/nf_templates/8_components.md (hunk 12 above)
Comment (christopher-hakkaart) [reply docs/nf_templates/8_components.md:117]: Thanks!

File: docs/nf_templates/7_testing.md (hunk 2 above)
[the remark this answers is not in the document]
Comment (christopher-hakkaart) [reply docs/nf_templates/7_testing.md:20]: I think this is a GitPod issue - it's working on my laptop. @ewels do you have any ideas?

**Comment by christopher-hakkaart** [comment]
Weird Netlify errors - will make a new PR

**Pull request closed without merging**